File: homework/quang_tu/lesson_1/sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int l = nums.size() - 1;

        for (int i = l, bound = i; i >= 0; --i) {
            if (nums[i] & 1) {
                int term = nums[bound];
                nums[bound] = nums[i];
                nums[i] = term;
                bound--;
            }
        }

        return nums;
    }
};

File: homework/Kiet_Tran/lesson_1/sort_array_by_parity.php
<?php

class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer[]
     */
    function sortArrayByParity(array $nums): array {
        $result = [];

        for ($i = 0; $i < count($nums); $i++) {
            if ($nums[$i] % 2 == 0) {
                $result[] = $nums[$i];
            }
        }

        for ($i = 0; $i < count($nums); $i++) {
            if ($nums[$i] % 2 != 0) {
                $result[] = $nums[$i];
            }
        }

        return $result;
    }
}
